Framework/Parallel/src/IO/MultiProcessEventLoader.cpp

```diff
@@ -0,0 +1,295 @@
+          command += std::to_string(evPerPr * i) + " "; // first event to load
+          command += std::to_string(upperBound) + " ";  // upper bound to load
+          command += std::to_string(m_numPixels) + " "; // pixel count
+          command += std::to_string(storageSize) + " "; // memory size
+          command += filename + " ";                    // nexus file name
+          command += groupname + " ";                   // instrument group name
+          command += m_precalculateEvents
+                         ? "1 "
+                         : "0 "; // variant of algorithm used for loading
+          for (unsigned j = 0; j < bankNames.size(); ++j) {
+            command += bankNames[j] + " ";                   // bank name
+            command += std::to_string(bankOffsets[j]) + " "; // bank size
+          }
+
+          try {
+            // launch child processes
+            vChilds.emplace_back(command.c_str());
+          } catch (std::exception const &ex) {
+            std::rethrow_if_nested(ex);
+          }
+        }
+    */
+    std::vector<Poco::ProcessHandle> vChilds;
+    for (unsigned i = 1; i < m_numProcesses; ++i) {
+      std::size_t upperBound =
+          i < m_numProcesses - 1 ? evPerPr * (i + 1) : numEvents;
+      std::vector<std::string> processArgs;
+
+      processArgs.push_back(m_segmentNames[i]);           // segment name
+      processArgs.push_back(m_storageName);               // storage name
+      processArgs.push_back(std::to_string(i));           // proc id
+      processArgs.push_back(std::to_string(evPerPr * i)); // first event to load
+      processArgs.push_back(std::to_string(upperBound));  // upper bound to load
+      processArgs.push_back(std::to_string(m_numPixels)); // pixel count
+      processArgs.push_back(std::to_string(storageSize)); // memory size
+      processArgs.push_back(filename);                    // nexus file name
+      processArgs.push_back(groupname); // instrument group name
+      processArgs.push_back(
+          m_precalculateEvents ? "1 "
+                               : "0 "); // variant of algorithm used for loading
+      for (unsigned j = 0; j < bankNames.size(); ++j) {
+        processArgs.push_back(bankNames[j]);                   // bank name
+        processArgs.push_back(std::to_string(bankOffsets[j])); // bank size
+      }
+
+      try {
+        // launch child processes
+        vChilds.emplace_back(
+            Poco::Process::launch(m_binaryToLaunch, processArgs));
+      } catch (std::exception const &ex) {
+        std::rethrow_if_nested(ex);
+      }
+    }
+
+    // Do self piece of multiprocess work
+    EventsListsShmemStorage storage(m_segmentNames[0], m_storageName,
+                                    storageSize, 1, m_numPixels, false);
+    fillFromFile(storage, filename, groupname, bankNames, bankOffsets, 0,
+                 numEvents / m_numProcesses, m_precalculateEvents);
```

Comment:
`evPerPr`.

In general, do you intend to automatically adjust the number of used processes depending on the file size?

Reply:
Yes, I do, but need some research.